test/c_src/finest.cpp

```diff
@@ -422,6 +422,52 @@ std::nullopt_t shared_mutex_shared_lock_test(ErlNifEnv *) {
 }
 FINE_NIF(shared_mutex_shared_lock_test, 0);
 
+std::nullopt_t condition_variable_test(ErlNifEnv *) {
+  bool notified = false;
+  fine::Mutex mutex("finest", "condition_variable_test_mutex");
+  fine::ConditionVariable cond("condition_variable_test");
+
+  std::thread thread1([&] {
+    {
+      auto lock = std::unique_lock{mutex};
+      cond.wait(lock, [&]() -> bool { return notified; });
+    }
+
+    cond.notify_all();
+  });
+
+  std::thread thread2([&] {
+    {
+      auto lock = std::unique_lock{mutex};
+      cond.wait(lock, [&]() -> bool { return notified; });
+    }
+
+    cond.notify_all();
+  });
+
+  std::thread thread3([&] {
+    {
+      auto lock = std::unique_lock{mutex};
+      cond.wait(lock, [&]() { return notified; });
+    }
+
+    cond.notify_all();
+  });
+
+  std::thread thread4([&] {
+    notified = true;
+    cond.notify_one();
```

Comment:
Theoretically thread4 could run first and then none of the other threads needs to wait, because the condition is already true.

We can make the test more robust, by having a `num_threads_counter` and each thread after obtaining the lock does `num_threads_counter++`. Then `thread4` can wait for `num_threads_counter == 4`, using another condition variable. wdyt?